src/Bicep.Core/TypeSystem/ResourceTypeRegistrar.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Azrm;
+
+namespace Bicep.Core.TypeSystem
+{
+    public class ResourceTypeRegistrar
+    {
+        public static ResourceTypeRegistrar Instance { get; } = new ResourceTypeRegistrar();
+
+        private IResourceTypeProvider resourceTypeProvider;
+
+        private ResourceTypeRegistrar()
+        {
+            resourceTypeProvider = new AzrmResourceTypeProvider();
```

Comment:
>resourceTypeProvider [](start = 12, length = 20)

I think we will probably need to have a discussion at some point how these providers get registered. Totally fine with hard-coding for now.